barretenberg/cpp/src/barretenberg/stdlib/primitives/field/field_conversion.test.cpp

```diff
@@ -12,34 +12,22 @@ template <typename Builder> class StdlibFieldConversionTests : public ::testing:
   public:
     template <typename T> void check_conversion(Builder& builder, T x)
     {
-        size_t len = bb::stdlib::field_conversion::calc_num_bn254_frs<T>();
-        auto frs = bb::stdlib::field_conversion::convert_to_bn254_frs(x);
+        size_t len = bb::stdlib::field_conversion::calc_num_bn254_frs<Builder, T>();
+        auto frs = bb::stdlib::field_conversion::convert_to_bn254_frs<Builder, T>(x);
         EXPECT_EQ(len, frs.size());
         auto y = bb::stdlib::field_conversion::convert_from_bn254_frs<Builder, T>(builder, frs);
         EXPECT_EQ(x.get_value(), y.get_value());
     }
 
-    template <typename T> void check_conversion_array(Builder& builder, T x)
+    template <typename T> void check_conversion_iterable(Builder& builder, T x)
     {
-        size_t len = bb::stdlib::field_conversion::calc_num_bn254_frs<T>();
-        auto frs = bb::stdlib::field_conversion::convert_to_bn254_frs(x);
+        size_t len = bb::stdlib::field_conversion::calc_num_bn254_frs<Builder, T>();
+        auto frs = bb::stdlib::field_conversion::convert_to_bn254_frs<Builder, T>(x);
         EXPECT_EQ(len, frs.size());
         auto y = bb::stdlib::field_conversion::convert_from_bn254_frs<Builder, T>(builder, frs);
         EXPECT_EQ(x.size(), y.size());
-        for (size_t i = 0; i < x.size(); i++) {
-            EXPECT_EQ(x[i].get_value(), y[i].get_value());
-        }
-    }
-
-    template <typename T> void check_conversion_univariate(Builder& builder, T x)
-    {
-        size_t len = bb::stdlib::field_conversion::calc_num_bn254_frs<T>();
-        auto frs = bb::stdlib::field_conversion::convert_to_bn254_frs(x);
-        EXPECT_EQ(len, frs.size());
-        auto y = bb::stdlib::field_conversion::convert_from_bn254_frs<Builder, T>(builder, frs);
-        EXPECT_EQ(x.evaluations.size(), y.evaluations.size());
-        for (size_t i = 0; i < x.evaluations.size(); i++) {
-            EXPECT_EQ(x.evaluations[i].get_value(), y.evaluations[i].get_value());
+        for (auto itx = x.begin(), ity = y.begin(); itx != x.end(); itx++, ity++) {
```

Comment:
You should be able to use our `zip` here, no?

Reply:
yep, updated.